src/fanControl.ino

```diff
@@ -0,0 +1,279 @@
+  pinMode(PIN_RELAY_6, OUTPUT);    
+  Serial.begin(115200);
+  //Serial.println("PWM 0 ready");
+  pwm.begin();
+  /*
+   * In theory the internal oscillator (clock) is 25MHz but it really isn't
+   * that precise. You can 'calibrate' this by tweaking this number until
+   * you get the PWM update frequency you're expecting!
+   * The int.osc. for the PCA9685 chip is a range between about 23-27MHz and
+   * is used for calculating things like writeMicroseconds()
+   * Analog servos run at ~50 Hz updates, It is importaint to use an
+   * oscilloscope in setting the int.osc frequency for the I2C PCA9685 chip.
+   * 1) Attach the oscilloscope to one of the PWM signal pins and ground on
+   *    the I2C PCA9685 chip you are setting the value for.
+   * 2) Adjust setOscillatorFrequency() until the PWM update frequency is the
+   *    expected value (50Hz for most ESCs)
+   * Setting the value here is specific to each individual I2C PCA9685 chip and
+   * affects the calculations for the PWM update frequency. 
+   * Failure to correctly set the int.osc value will cause unexpected PWM results
+   */
+  pwm.setOscillatorFrequency(27000000);
+  pwm.setPWMFreq(1600);  // This is the maximum PWM frequency
+
+  // if you want to really speed stuff up, you can go into 'fast 400khz I2C' mode
+  // some i2c devices dont like this so much so if you're sharing the bus, watch
+  // out for this!
+  Wire.setClock(400000);
+}
+
+void loop() 
+{
+  int i = 0;
+  while (Serial.available() > 0) 
+  {
+    i = i + Serial.readBytes(inBuf, 4);
+//    i++;
+//    i = 4;
+    if (i == 4) 
+    {
+      i = 0;
+//      Serial.println("4 bytes read!");
+      if (inBuf[3] == CMD_FAN_ID)
+      {
+//        Serial.println("CMD_FAN_ID");
+        // Send Fan ID
+        outBuf[3] = CMD_FAN_ID;
+        outBuf[2] = FAN_ID;
+        outBuf[1] = 0;
+        outBuf[0] = CMD_OK;
+      }
+      else if (inBuf[3] == CMD_RELAY)
+      {
+//        Serial.println("CMD_RELAY");
+        outBuf[3] = CMD_RELAY;
+        outBuf[2] = inBuf[2]; // channel
+        outBuf[1] = inBuf[1];  // relay set/reset
+        outBuf[0] = CMD_OK;
+        if ((inBuf[1] == RELAY_SET) || (inBuf[1] == RELAY_RESET))
+        {
+          if ((inBuf[2] < 14) && (inBuf[2] > 7))   // channel no.
+          {
+            digitalWrite(inBuf[2], inBuf[1]);
+          }
+          else
+          {
+            outBuf[0] = CMD_ERROR;        
+          }
+        }
+        else
+        {
+          outBuf[0] = CMD_ERROR;
+        }
+      }
+      else if (inBuf[3] == CMD_SPEED)
+      {
+  //      Serial.println("CMD_SPEED");
+        noInterrupts(); // I think we need this to successfully copy the input
+        channel = inBuf[2];
+        duty = inBuf[1];
+        interrupts();
+        outBuf[3] = CMD_SPEED;
+        outBuf[2] = channel;
+        outBuf[1] = duty;
+        outBuf[0] = CMD_OK;
+
+        if (channel < 16) // There are 16 channels on the Adafruit PWM board

```

Comment:
Shouldn't channel also be between 8 and 13? Just like in `CMD_RELAY`.

Reply:
The channel number refers to the PWM shield channel number,

Reply:
But does an `inBuff[2]` of say 9 still means `PIN_RELAY_2`, or do the 16 channels of the shield mean something else all together? 
I mean, do `channel` values that are smaller than 8 or greater than 13 have any meaning, implications, or side-effects at all?

Reply:
My idea was that you address the fan groups from the PC, always 1,2,3,4,5. On the Arduino side Relay settings go through I/O pins 8,9,10,11,12 and Fan speed through PWM shield channel 0,1,2,3,4, but the PC side of the code should not bother about the difference I suppose. Makes it easier to change I/Os or PWM:s on the Arduino side without messing up the PC side. Or am I getting you wrong?